rust/candid/src/types/impls.rs

```diff
@@ -260,3 +260,52 @@ tuple_impls! {
     15 => (0 T0 1 T1 2 T2 3 T3 4 T4 5 T5 6 T6 7 T7 8 T8 9 T9 10 T10 11 T11 12 T12 13 T13 14 T14)
     16 => (0 T0 1 T1 2 T2 3 T3 4 T4 5 T5 6 T6 7 T7 8 T8 9 T9 10 T10 11 T11 12 T12 13 T13 14 T14 15 T15)
 }
+
+impl CandidType for std::time::SystemTime {
+    fn id() -> TypeId {
+        TypeId::of::<std::time::SystemTime>()
+    }
+
+    fn _ty() -> Type {
+        Type::Record(vec![
+            Field {
+                id: Label::Named("nanos_since_epoch".to_owned()),
+                ty: u32::_ty(),
```

Comment:
```suggestion
                ty: u32::ty(),
```

Reply:
`ty()` is the memoized version of `_ty()` to handle recursion. It won't make a difference here, but generally good coding style.